consoles/VNC.pm

```diff
@@ -921,6 +934,44 @@ sub _receive_zrle_encoding ($self, $x, $y, $w, $h) {
     return $res;
 }
 
+sub _receive_tight_encoding ($self, $x, $y, $w, $h) {
+    my $socket = $self->socket;
+
+    my $pi = $self->_pixinfo;
+
+    my $stime = time;
+    $socket->read(my $data, 1)
+      or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for compression control');
+    my ($compression_control) = unpack('C', $data);
+    # Only JPEG for now
+    die "Unsupported compression $compression_control" if ($compression_control & 0xF0) != 0x90;
+    $socket->read($data, 1)
+      or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for data len');
+    my ($data_len) = unpack('C', $data);
+    if ($data_len & 0x80) {
+        $socket->read($data, 1)
+          or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for data len');
+        my ($data_len2) = unpack('C', $data);
+        $data_len = ($data_len & 0x7f) | ($data_len2 & 0x7f) << 7;
+        if ($data_len2 & 0x80) {
+            $socket->read($data, 1)
+              or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for data len');
+            my ($data_len2) = unpack('C', $data);
+            $data_len |= $data_len2 << 14;
+        }
+    }
+
+    my $read_len = 0;
+    while ($read_len < $data_len) {
+        my $len = read($socket, $data, $data_len - $read_len, $read_len);
+        OpenQA::Exception::VNCProtocolError->throw(error => "short read for zrle data $read_len - $data_len") unless $len;
+        $read_len += $len;
+    }
+    diag sprintf("read $data_len in %fs\n", time - $stime) if (time - $stime > 0.1);
+    my $image = tinycv::from_ppm($data);
```

Comment:
So `from_ppm` can actually do JPEG decoding? Or where does the decoding happen?

Reply:
`imdecode()`, which is more or less `imread()` [supports various formats, including PPM, PNG and JPEG](https://docs.opencv.org/4.8.0/d4/da8/group__imgcodecs.html#ga288b8b3da0892bd651fce07b3bbd3a56)
> The function determines the type of an image by the content, not by the file extension.

Reply:
Ah, so the function name `image_from_ppm` we have chosen for our bindings is actually a bit misleading.